File: aggregator/src/main/java/com/web3analytics/models/MevAlert.java
package com.web3analytics.models;

import java.util.List;

/**
 * MEV (Maximal Extractable Value) alert detected from cross-event pattern analysis.
 * Produced by session windows that group same-block activity per pair.
 */
public record MevAlert(
        String alertId,
        String alertType,
        long windowStart,
        long windowEnd,
        String pairAddress,
        String token0Symbol,
        String token1Symbol,
        long blockNumber,
        String attackerAddress,
        List<String> victimAddresses,
        double estimatedProfitUSD,
        int involvedSwapCount,
        int involvedEventCount,
        String severity,
        String description,
        List<String> involvedTransactions,
        long detectedAt
) {}

File: analytics-service/src/main/kotlin/com/web3analytics/analytics/model/MevAlert.kt
package com.web3analytics.analytics.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class MevAlert(
    val alertId: String,
    val alertType: String,
    val windowStart: Long = 0,
    val windowEnd: Long = 0,
    val pairAddress: PairAddress,
    val token0Symbol: String? = null,
    val token1Symbol: String? = null,
    val blockNumber: Long = 0,
    val attackerAddress: String? = null,
    val victimAddresses: List<String> = emptyList(),
    val estimatedProfitUSD: Double = 0.0,
    val involvedSwapCount: Int = 0,
    val involvedEventCount: Int = 0,
    val severity: String = "LOW",
    val description: String? = null,
    val involvedTransactions: List<String> = emptyList(),
    val detectedAt: Long = 0
)
